src/index.js

```diff
@@ -1 +1,118 @@
 'use strict';
+
+const http = require('http');
+const fs = require('fs');
+const formidable = require('formidable');
+
+const PORT = process.env.PORT || 3000;
+const jsonFile = './src/expence.json';
+
+const server = new http.Server();
+
+server.on('request', (req, res) => {
+  const normilizedUrl = new URL(req.url, `http://${req.headers.host}`);
+  const pathName = normilizedUrl.pathname.slice(1) || 'index.html';
+
+  if (pathName === 'index.html') {
+    res.setHeader('Content-Type', 'text/html');
+
+    const readData = fs.createReadStream(`./src/public/${pathName}`);
+
+    readData.on('error', () => {
+      res.statusCode = '400';
+      res.end();
+    });
+
+    readData.pipe(res);
+
+    res.on('close', () => {
+      readData.destroy();
+    });
+
+    return;
+  }
+
+  if (pathName === 'expense') {
+    const form = new formidable.IncomingForm();
+
+    form.parse(req, (error, fields) => {
+      if (error) {
+        res.statusCode = '500';
+        res.end();
+      }
+
+      const { date, title, amount } = fields;
+
+      const jsonFields = JSON.stringify(fields);
+
+      fs.writeFileSync(jsonFile, jsonFields);
```

Comment:
You are using 'fs.writeFileSync' which is a synchronous method. It is recommended to use asynchronous methods in node.js to avoid blocking the event loop. You can use 'fs.writeFile' instead and handle the callback.